packages/web-ui/src/ds/atoms/Modal/index.tsx

```diff
@@ -37,10 +48,65 @@ export function Modal({
           </DialogHeader>
         ) : null}
 
-        <div className='py-4'>{children}</div>
+        {children}
 
         {footer ? <DialogFooter>{footer}</DialogFooter> : null}
       </DialogContent>
     </Dialog>
   )
 }
+
+type ConfirmModalProps = Omit<ModalProps, 'footer' | 'children'> & {
+  type?: 'default' | 'destructive'
+  confirm: {
+    label: string
+    title?: string
+    isConfirming?: boolean
+    description?: string
+  }
+  cancel?: { label?: string }
+  onConfirm: () => void
+  onCancel?: () => void
+  children?: ReactNode
+}
+export function ConfirmModal({
+  type = 'default',
+  confirm,
+  cancel,
+  onConfirm,
+  onCancel,
+  children,
+  ...rest
+}: ConfirmModalProps) {
+  return (
+    <Modal
+      {...rest}
+      footer={
+        <div className='flex justify-end space-x-2'>
+          <DialogClose asChild>
+            <Button variant='outline' onClick={onCancel}>
+              {cancel?.label ?? 'Cancel'}
+            </Button>
+          </DialogClose>
+          <Button
+            isLoading={confirm.isConfirming}
+            variant={type}
+            onClick={onConfirm}
+          >
+            {confirm.label}
+          </Button>
+        </div>
+      }
+    >
+      {confirm.description || confirm.title ? (
```

Comment:
add parenthesis to make clear the order of evaluation, not sure if prettier is gonna remove them though

Reply:
Prettier removed the parenthesis